controller/kubernetes_pod_controller.go

```diff
@@ -493,6 +505,11 @@ func (kc *KubernetesPodController) handlePodDeletionIfVolumeRequestRemount(pod *
 		if vol.Status.RemountRequestedAt == "" {
 			continue
 		}
+
+		if isRegularRWXVolume(vol) && !isStorageNetworkForRWXVolume {
```

Comment:
NIT: It is obvious why in context, but can we add a comment here explaining the logic? The following as an example, or something better?

```suggestion
		// NFS clients can generally recover without a remount when the NFS server restarts using the same ClusterIP. 
		// A remount is required when a storage network is in use because the new NFS server has a different IP.
		if isRegularRWXVolume(vol) && !isStorageNetworkForRWXVolume {
```

Reply:
Done.